gitlab/v4/objects/repositories.py

```diff
@@ -246,6 +246,32 @@ def repository_archive(
             result, streamed, action, chunk_size, iterator=iterator
         )
 
+    @cli.register_custom_action("Project", ("refs",))
+    @exc.on_http_error(exc.GitlabGetError)
+    def repository_merge_base(
+        self, refs: List[str], **kwargs: Any
+    ) -> Union[Dict[str, Any], requests.Response]:
```

Comment:
This made me think if we should start forcing these to just be a `Dict`?

Reply:
Thanks @JohnVillalovos! Makes sense to me! I was looking at doing this via overloading `http_get` to narrow the types there already, depending on the streamed/raw arguments but it got a bit out of hand. Then I think we wouldn't need to do it in "downstream" methods, IIUC. I will look into it again :)

But it requires quite a few overloads for all the possible signatures. This is what I was looking at https://medium.com/analytics-vidhya/making-sense-of-typing-overload-437e6deecade.